state-recover/besu-plugin/src/main/kotlin/linea/staterecover/plugin/RecoveryModeManager.kt

```diff
@@ -0,0 +1,103 @@
+package linea.staterecover.plugin
+
+import linea.staterecover.RecoveryStatusPersistence
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.hyperledger.besu.plugin.data.AddedBlockContext
+import org.hyperledger.besu.plugin.services.BesuEvents
+import org.hyperledger.besu.plugin.services.mining.MiningService
+import org.hyperledger.besu.plugin.services.p2p.P2PService
+import org.hyperledger.besu.plugin.services.sync.SynchronizationService
+import java.util.concurrent.atomic.AtomicBoolean
+
+class RecoveryModeManager(
```

Comment:
Just to make sure I understand, from Genesis block until block `targetBlockNumber` the node is expected to sync as usual, without L1 data. Is this correct?

Reply:
Yes, that is correct.